src/Controls/tests/Xaml.UnitTests/Issues/Maui21434.xaml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+                    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+                    xmlns:local="clr-namespace:Microsoft.Maui.Controls.Xaml.UnitTests"
+                    x:Class="Microsoft.Maui.Controls.Xaml.UnitTests.Maui21434"
+                    x:DataType="local:ParentViewModel21434">
+    <ContentPage.BindingContext>
+        <local:ParentViewModel21434 />
+    </ContentPage.BindingContext>
+
+    <VerticalStackLayout>
+        <Label Text="{Binding Text}" x:Name="ParentTextLabel" />
+        <Label BindingContext="{Binding Child}" Text="{Binding Text}" x:DataType="local:ChildViewModel21434" x:Name="ChildTextLabel" />
```

Comment:
could we be even smarter, and infer the x:DataType of the label ? not in all cases, but in this one it looks like we could...

Reply:
I think we could, I opened a new issue so that we can discuss all the cases when we could infer x:DataTypes (https://github.com/dotnet/maui/issues/21834)